docs/documentation.md

```diff
@@ -0,0 +1,421 @@
+
+
+## Enabling creator royalties for NFTs
+
+The `NFTStorefrontV2` contract supports royalties, but it doesn't mandate them. It is a choice of marketplaces whether they want to support the creator royalties during the listing or not. However, we encourage the marketplaces to support the royalties to stimulate the artist's participation in the **FLOW** ecosystem.
+
+If seller's NFT supports [Royalty Metadata View](https://github.com/onflow/flow-nft/blob/21c254438910c8a4b5843beda3df20e4e2559625/contracts/MetadataViews.cdc#L335) standard, then marketplaces can support royalties during the fulfilment of a listing. `NFTStorefrontV2` can dynamically fetch the royalty details during the listing creation and turn them into the `saleCut` of the listing.
+
+```cadence
+// Check whether the NFT implements the MetadataResolver or not.
+if nft.getViews().contains(Type<MetadataViews.Royalties>()) {
+		// Resolve the royalty view
+    let royaltiesRef = nft.resolveView(Type<MetadataViews.Royalties>())?? panic("Unable to retrieve the royalties")
+	  // Fetch the royalties.
+		let royalties = (royaltiesRef as! MetadataViews.Royalties).getRoyalties()
+		// Append the royalties as the salecut
+    for royalty in royalties {
+        self.saleCuts.append(NFTStorefrontV2.SaleCut(receiver: royalty.receiver, amount: royalty.cut * effectiveSaleItemPrice))
+        totalRoyaltyCut = totalRoyaltyCut + royalty.cut * effectiveSaleItemPrice
+    }
+}
+```
+
+Complete transaction can be viewed [here](https://github.com/onflow/nft-storefront/blob/main/transactions/sell_item.cdc).
+
+`saleCut` only supports a single receiver type because the beneficiary of saleCut cannot receive different currency as the royalty or a general sale percentage cut. To resolve this, we recommend using the [FungibleTokenSwitchboard](https://github.com/onflow/flow-ft/blob/master/contracts/FungibleTokenSwitchboard.cdc) contract. It defines a generic receiver for fungible tokens so a user can always provide their generic receiver, regardless of what fungible token they want. The switchboard will manage the routing of funds to the respective vault. Read more about this [here](https://github.com/onflow/flow-ft#fungible-token-switchboard).
+
+## Enabling marketplace commissions for NFT sales
+
+`NFTStorefrontV2` introduces the commissions on trade to incentivize the listing's facilitator (i.e Marketplaces , dapps). Commission & commission receivers are set during the listing creation, and only one of the commission receivers, i.e. `marketplaceAddress` can take away to commission after the fulfilment of the listing. However, if a user wants to create a p2p listing, then the user can set commission receivers as `nil`. At the same time, the commission will be available to anyone for grabs. If the buyer of the NFT and `commissionRecipient` is the same, then the commission can act as the discount.
```

Comment:
"`NFTStorefrontV2` enables optional commissions on trades for marketplaces which require it as a condition to list a NFT for sale. Commission & commission receivers are set by the seller during initial listing creation. At time of purchase the commission amount is paid once only to the commission receiver matching the marketplace receiver address which facilitated the sale.

For NFT listings in marketplaces which don't require commission, commission receivers can be set as `nil`. Setting the buyer of the NFT and `commissionRecipient` to the same has the effect of applying a discount for the buyer."

@satyamakgec from the original text can you please explain what the specific way that this would work when commission is 'up for grabs' - "At the same time, the commission will be available to anyone for grabs". it's not clear how it works, who gets the commission and how (doesn't a receiver address need to first be set?)